test/screens/result/result_screen_test.dart

```diff
@@ -0,0 +1,315 @@
+          rate,
+        ),
+      ),
+      findsOneWidget,
+    );
+
+    expect(
+      find.byWidgetPredicate(
+        (widget) => findRowWithTitleAndDescription(
+          widget,
+          'Rentabilidade anual',
+          annualGrossRateProfit,
+        ),
+      ),
+      findsOneWidget,
+    );
+
+    expect(
+      find.byWidgetPredicate(
+        (widget) => findRowWithTitleAndDescription(
+          widget,
+          'Rentabilidade no período',
+          rateProfit,
+        ),
+      ),
+      findsOneWidget,
+    );
+
+    final button = find.byWidgetPredicate(
+      (widget) => findButtonWithText(
+        widget,
+        'Simular novamente',
+        isEnabled: true,
+      ),
+    );
+    expect(button, findsOneWidget);
+  });
+
+  testWidgets('when request fails', (tester) async {
+    when(repository.doSimulation(any)).thenThrow(Exception('error'));
+
+    await setupContainer(
+      tester,
+      ResultContainer(
+        repository: repository,
+        onBack: expectAsync1((_) {}, count: 0),
+      ),
+    );
+
+    expect(find.widgetWithText(AppBar, 'Resultado'), findsOneWidget);
+    expect(find.text('Oooops...'), findsOneWidget);
+    expect(
+      find.text('Algo não saiu como planejado e as\n informações não foram'
+          ' carregadas.'),
+      findsOneWidget,
+    );
+
+    final button = find.byWidgetPredicate(
+      (widget) => findButtonWithText(
+        widget,
+        'Tentar novamente',
+        isEnabled: true,
+      ),
+    );
+    expect(button, findsOneWidget);
+  });
+
+  testWidgets('when request fails and click on try again returns success',
+      (tester) async {
+    when(repository.doSimulation(any)).thenThrow(Exception('error'));
+
+    await setupContainer(
+      tester,
+      ResultContainer(
+        repository: repository,
+        onBack: expectAsync1((_) {}, count: 0),
+      ),
+    );
+
+    expect(find.byType(ResultErrorScreen), findsOneWidget);
+
+    when(repository.doSimulation(any)).thenAnswer((_) async => result);
+
+    final button = find.byWidgetPredicate(
+      (widget) => findButtonWithText(
+        widget,
+        'Tentar novamente',
+        isEnabled: true,
+      ),
+    );
+    await tester.tap(button);
+    await tester.pumpAndSettle();
+
+    expect(find.byType(ResultSuccessScreen), findsOneWidget);
+  });
+
+  testWidgets('when simulate again button is clicked', (tester) async {
+    when(repository.doSimulation(any)).thenAnswer((_) async => result);
+
+    await setupContainer(
+      tester,
+      ResultContainer(
+        repository: repository,
+        onBack: expectAsync1((_) {}, count: 0),
+      ),
+    );
+
+    final button = find.byWidgetPredicate(
+      (widget) => findButtonWithText(
+        widget,
+        'Simular novamente',
+        isEnabled: true,
+      ),
+    );
+    await tester.tap(button);
+    await tester.pumpAndSettle();
+  });
+
+  testWidgets('when back button is clicked', (tester) async {
```

Comment:
@williamhjcho por algum motivo o teste falha quando clico no voltar. Saberia dizer o motivo?

> `══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown while handling a gesture:
Closure call with mismatched arguments: function 'main.<anonymous closure>.<anonymous closure>'
Receiver: Closure: (Object?) => void
Tried calling: main.<anonymous closure>.<anonymous closure>()
Found: main.<anonymous closure>.<anonymous closure>(Object?) => void
When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:54:5)
#1      _objectNoSuchMethod (dart:core-patch/object_patch.dart:97:9)
#2      Function._apply (dart:core-patch/function_patch.dart:11:30)
#3      Function.apply (dart:core-patch/function_patch.dart:34:12)
#4      _ExpectedFunction._run (package:test_api/src/frontend/expect_async.dart:193:23)
#5      _ExpectedFunction.max6 (package:test_api/src/frontend/expect_async.dart:176:7)
#6      _ExpectedFunction.max1 (package:test_api/src/frontend/expect_async.dart:144:41)
#7      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:989:21)
#8      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:182:24)
#9      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:607:11)
#10     BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:296:5)
#11     BaseTapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:230:7)
#12     PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:475:9)
#13     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:93:12)
#14     PointerRouter._dispatchEventToRoutes.<anonymous closure> (package:flutter/src/gestures/pointer_router.dart:138:9)
#15     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:397:8)
#16     PointerRouter._dispatchEventToRoutes (package:flutter/src/gestures/pointer_router.dart:136:18)
#17     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:122:7)
#18     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:439:19)
#19     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:419:22)
#20     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:287:11)
#21     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:374:7)
#22     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:338:5)
#23     TestWidgetsFlutterBinding.handlePointerEvent (package:flutter_test/src/binding.dart:507:13)
#24     WidgetTester.sendEventToBinding.<anonymous closure> (package:flutter_test/src/widget_tester.dart:792:15)
#25     WidgetTester.sendEventToBinding.<anonymous closure> (package:flutter_test/src/widget_tester.dart:791:39)
#28     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:71:41)
#29     WidgetTester.sendEventToBinding (package:flutter_test/src/widget_tester.dart:791:27)
#30     TestGesture.up.<anonymous closure> (package:flutter_test/src/test_pointer.dart:392:24)
#31     TestGesture.up.<anonymous closure> (package:flutter_test/src/test_pointer.dart:390:39)
#34     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:71:41)
#35     TestGesture.up (package:flutter_test/src/test_pointer.dart:390:27)
#36     WidgetController.tapAt.<anonymous closure> (package:flutter_test/src/controller.dart:278:21)
<asynchronous suspension>
<asynchronous suspension>(elided 5 frames from dart:async and package:stack_trace)
Handler: "onTap"
Recognizer:
  TapGestureRecognizer#84d81`

Reply:
puts, verdade. Como o `onBack` é um `VoidCallback` o correto seria utilizar o `expectAsync0(() {}, count: 1)` ao invés do `expectAsync1((_) {}, count: 1)`. Boa!